packages/components/src/ChartErrorOverlay.tsx

```diff
@@ -0,0 +1,51 @@
+import { ReactElement } from 'react';
+import classNames from 'classnames';
+import { ButtonOld } from '.';
+import './ChartErrorOverlay.scss';
+
+interface ChartErrorOverlayProps {
+  errorMessage: string;
+  clearError?: () => void;
+  'data-testid'?: string;
+}
+
+function ChartErrorOverlay({
+  errorMessage,
+  clearError,
+  'data-testid': dataTestId,
+}: ChartErrorOverlayProps): ReactElement {
+  const messageTestId =
+    dataTestId != null ? `${dataTestId}-message` : undefined;
+
+  const slowPerformanceMessage = errorMessage.includes(
+    'Plot contains more than'
+  ); // TODO: core side changes to detect slow performance and show this message
+
+  const unableToDownsampleMessage =
+    errorMessage === 'Too many items to disable downsampling';
+
+  const undismissableError = unableToDownsampleMessage;
+
+  return (
+    <div className="chart-panel-overlay chart-error-overlay">
+      <div
+        className={classNames(
+          'chart-panel-overlay-content chart-error-overlay-content'
+        )}
+      >
```

Comment:
All of these class names are specific to the charting plugin, which is in the `dashboard-core-plugins` package. The `components` package (which this is currently in) should be generic components and should not depend on `dashboard-core-plugins`. This file should probably be in `dashboard-core-plugins` or `chart`